kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1522 @@
+            #pragma omp parallel firstprivate(transformation_matrix, constant_vector, slave_dof_list, master_dof_list, aux_solution_vector, aux_master_vector)
+            {
+                // The current process info
+                ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
+
+                #pragma omp for schedule(guided, 512)
+                for (int i_const = 0; i_const < number_of_constraints; ++i_const) {
+                    auto it_const = rModelPart.MasterSlaveConstraints().begin() + i_const;
+
+                    // Detect if the constraint is active or not. If the user did not make any choice the constraint
+                    // It is active by default
+                    bool constraint_is_active = true;
+                    if (it_const->IsDefined(ACTIVE))
+                        constraint_is_active = it_const->Is(ACTIVE);
+
+                    if (constraint_is_active) {
+                        it_const->CalculateLocalSystem(transformation_matrix, constant_vector, r_current_process_info);
+                        it_const->GetDofList(slave_dof_list, master_dof_list, r_current_process_info);
+
+                        // The direct assignment will be done in case all the dofs of the constraint are fixed
+                        // Otherwise the behaviour is undetermined
+                        // How do you impose directly at the same time you solve the system?, good question
+                        SizeType number_dofs = 0;
+                        for (auto& master_dof : master_dof_list) {
+                            if (master_dof->IsFixed()) {
+                                ++number_dofs;
+                            }
+                        }
+
+                        // Now we add the contribution
+                        if (number_dofs == master_dof_list.size()) {
+                            aux_solution_vector.resize(slave_dof_list.size());
+                            aux_master_vector.resize(number_dofs);
+
+                            IndexType counter = 0;
+                            for (auto& master_dof : master_dof_list) {
+                                aux_master_vector[counter] = master_dof->GetSolutionStepValue();
+                                ++counter;
+                            }
+                            noalias(aux_solution_vector) = prod(transformation_matrix, aux_master_vector);
+
+                            counter = 0;
+                            for (auto& slave_dof : slave_dof_list) {
+                                const IndexType node_id = slave_dof->Id();
+                                const auto& r_variable = slave_dof->GetVariable();
+                                auto pnode = rModelPart.pGetNode(node_id);
+                                double coeff;
+                                if (r_variable.IsNotComponent()) {
+                                    coeff = pnode->GetValue(NODAL_MAUX);
```

Comment:
This operation I deactivated by default, and basically is because if a slave dofs is related with a fixed master dofs there is no other way to move it consistently